test/services/database/comment/comment_repository_test.dart

```diff
@@ -0,0 +1,316 @@
+      });
+
+      test(
+          "should initialize the comment count of a post to 1 when the commentCount field doesn't exist and a post is added",
+          () async {
+        // Remove the comment count field
+        await postDocument
+            .update({PostData.commentCountField: FieldValue.delete()});
+
+        final commentData = postCommentDataGenerator.createMockCommentData();
+
+        await commentRepo.addComment(
+          post.id,
+          commentData,
+        );
+
+        // Check that the comment count was updated correctly
+        final postDoc = await postDocument.get();
+        final actualPost = PostFirestore.fromDb(postDoc);
+
+        expect(actualPost.data.commentCount, equals(1));
+      });
+    });
+
+    /// Utility function to check that the post and user comments are not empty
+    Future<void> checkPostAndUserCommentsNotEmpty() async {
+      final postComments = await commentRepo.getPostComments(post.id);
+      final userComments = await commentRepo.getUserComments(user.uid);
+
+      expect(postComments, isNotEmpty);
+      expect(userComments, isNotEmpty);
+    }
+
+    /// Utility function to check that the post and user comments are empty
+    Future<void> checkPostAndUserCommentsEmpty() async {
+      final postComments = await commentRepo.getPostComments(post.id);
+      final userComments = await commentRepo.getUserComments(user.uid);
+
+      expect(postComments, isEmpty);
+      expect(userComments, isEmpty);
+    }
+
+    Future<(List<UserFirestore>, List<CommentFirestore>)> addCommentsForUsers(
+      int nbUsers,
+    ) async {
+      final users =
+          await FirestoreUserGenerator.addUsers(fakeFirestore, nbUsers);
+      final postComments = <CommentFirestore>[];
+
+      for (user in users) {
+        final commentData =
+            postCommentDataGenerator.createMockCommentData(ownerId: user.uid);
+
+        final commentId = await commentRepo.addComment(post.id, commentData);
+
+        postComments.add(CommentFirestore(id: commentId, data: commentData));
+      }
+
+      // Check that the user comments have been added
+      for (user in users) {
+        final userComments = await commentRepo.getUserComments(user.uid);
+
+        expect(userComments, isNotEmpty);
+      }
+
+      return (users, postComments);
+    }
+
+    group("deleting comments", () {
+      test("should delete a comment", () async {
+        // Add a comment
+        final commentData =
+            postCommentDataGenerator.createMockCommentData(ownerId: user.uid);
+
+        final commentId = await commentRepo.addComment(post.id, commentData);
+
+        // Check that it was added correctly
+        await checkPostAndUserCommentsNotEmpty();
+
+        // Delete the comment
+        await commentRepo.deleteComment(post.id, commentId, user.uid);
+
+        // Check that it was deleted correctly
+        await checkPostAndUserCommentsEmpty();
+      });
```

Comment:
I wonder if this test is really useful, considering the following test does the same but is more robust.

Reply:
Looking again at it, I don't think that it is necessary anymore.
Removed in 4e7d49eecd82658bef0ebbab76c9cbb3bacdd8a1.